rotkehlchen/chain/ethereum/modules/yearn/decoder.py

```diff
@@ -93,97 +107,131 @@ def reload_data(self) -> None:
-            context.event.notes = f'Withdraw {context.event.balance.amount} {context.token.symbol} from {protocol} vault {vault_token_name}'  # noqa: E501
+            event.event_type = HistoryEventType.DEPOSIT
+            event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+            event.counterparty = counterparty
+            event.notes = f'Deposit {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} in {counterparty} vault'  # noqa: E501
+            if event.address != YEARN_PARTNER_TRACKER:
+                vault_token_name = _get_vault_token_name(event.address)
+                event.notes = f'{event.notes} {vault_token_name}'
         elif (
-            is_deposit is False and
-            context.event.event_type == HistoryEventType.SPEND and
-            context.event.address == ZERO_ADDRESS
+            event.event_type == HistoryEventType.RECEIVE and
+            event.event_subtype == HistoryEventSubType.NONE and
+            event.address == ZERO_ADDRESS
         ):
-            context.event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
-            context.event.counterparty = protocol
-            context.event.notes = f'Return {context.event.balance.amount} {context.token.symbol} to a {protocol} vault'  # noqa: E501
-        else:
-            # in this case we failed to find a valid transfer event. Inform about the failure
-            return FAILED_ENRICHMENT_OUTPUT
-
-        return TransferEnrichmentOutput(matched_counterparty=protocol)
+            event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+            event.counterparty = counterparty
+            event.notes = f'Receive {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} after deposit in a {counterparty} vault'  # noqa: E501
+
+    def _decode_deposit_withdraw(self, context: DecoderContext) -> DecodingOutput:
+        """Decode deposit/withdraw events for yearn v2 and v3."""
+        for event in context.decoded_events:
+            if context.tx_log.topics[0] == YEARN_V2_WITHDRAW_TOPIC:
+                self._handle_withdraw_event(event=event, protocol=YEARN_VAULTS_V2_PROTOCOL)
+            elif context.tx_log.topics[0] == YEARN_V3_WITHDRAW_TOPIC:
+                self._handle_withdraw_event(event=event, protocol=YEARN_VAULTS_V3_PROTOCOL)
+            elif context.tx_log.topics[0] == YEARN_V2_DEPOSIT_TOPIC:
+                self._handle_deposit_event(event=event, protocol=YEARN_VAULTS_V2_PROTOCOL)
+            elif context.tx_log.topics[0] == YEARN_V3_DEPOSIT_TOPIC:
+                self._handle_deposit_event(event=event, protocol=YEARN_VAULTS_V3_PROTOCOL)
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_v2_increase_deposit(self, context: DecoderContext) -> DecodingOutput:
+        """Decode increase deposit events for yearn v2."""
+        if context.tx_log.topics[0] == YEARN_V2_INCREASE_DEPOSIT_TOPIC:
+            for event in context.decoded_events:
```

Comment:
Why do it for all decoded events? This does not make sense.

Reply:
Moved loop into subfunction, see https://github.com/rotki/rotki/pull/8795#discussion_r1813412944